src/vim9execute.c

```diff
@@ -2147,7 +2147,14 @@ execute_storeindex(isn_T *iptr, ectx_T *ectx)
     // -1 dict, list, blob or object
     tv = STACK_TV_BOT(-3);
     SOURCING_LNUM = iptr->isn_lnum;
-    if (dest_type == VAR_ANY)
+
+    // Make sure an object has been initialized
+    if (dest_type == VAR_OBJECT && tv_dest->vval.v_object == NULL)
```

Comment:
I think this check should be made a compile time check and not a runtime check (for performance reasons).

Reply:
I'm not sure it's possible, for example if the NULL object is the argument of a function, how do you check that at compile time? Like in
```vim
vim9script

class C
    static val: string
endclass

def B(o: obj)
    echo o.val
enddef

def F()
    var obj: C
    B(obj)
enddef
F()
```

Reply:
An alternative would be to disallow the initialization of an object without a constructor (that is, to always require `new()`) but I don't know how (if) it can be done, for example when the object is again the parameter of a function.

Reply:
A NULL check doesn't seem too taxing anyway at runtime.